pkg/csi/service/common/vsphereutil.go

```diff
@@ -827,6 +842,20 @@ func QueryVolumeSnapshotsByVolumeID(ctx context.Context, volManager cnsvolume.Ma
 		return nil, "", err
 	}
 	for _, queryResult := range queryResultEntries {
+		// check if the queryResult has fault, if so, the specific result can be ignored.
+		if queryResult.Error != nil {
+			// Currently, CnsVolumeNotFoundFault is the only possible fault when QuerySnapshots is
+			// invoked with only volume-id
+			fault := queryResult.Error.Fault
+			if faultInfo, ok := fault.(cnstypes.CnsVolumeNotFoundFault); ok {
+				faultVolumeId := faultInfo.VolumeId.Id
+				log.Warnf("volume %s was not found during QuerySnapshots, ignore volume..", faultVolumeId)
+			} else {
+				log.Errorf("unexpected fault %+v received in QuerySnapshots result: %+v", fault, queryResult)
+			}
```

Comment:
There could be other internal errors.  I think this unexpected error should not be ignored.

Reply:
Again, this is unlikely, the spec clearly states that the only possible error is `CnsVolumeNotFound` as per https://opengrok.eng.vmware.com/source/xref/main.perforce.1666/bora/vsan/pyvmodl/CnsVolumeManager.py#1962 , any other errors are unlikely, under the event we do get some unexpected error I feel we should process the other results, erroring out entirely seems like a overkill.

Reply:
Even though right now "not found" is the only error, it does not mean other errors won't be added in the future.  In that case, I think it is better to return error.  Can you think of any downside of returning an error?